h/api/models/test/document_test.py

```diff
@@ -119,6 +122,416 @@ def test_document_find_or_create_by_uris_no_results():
+        type='self-claim',
+        content_type=None,
+        document=mock_document(),
+        created=datetime.datetime.now(),
+        updated=datetime.datetime.now())
+
+    db.add.assert_called_once_with(DocumentURI.return_value)
+
+
+@create_or_update_document_uri_fixtures
+def test_create_or_update_document_uri_does_not_create_new_DocumentURI(DocumentURI):
+    """It shouldn't create a new DocumentURI if one already exists."""
+    DocumentURI.query.filter.return_value.first.return_value = mock.Mock()
+    db = mock_db()
+
+    document.create_or_update_document_uri(
+        db=db,
+        claimant='http://example.com/example_claimant.html',
+        uri='http://example.com/example_uri.html',
+        type='self-claim',
+        content_type=None,
+        document=mock_document(),
+        created=datetime.datetime.now(),
+        updated=datetime.datetime.now())
+
+    assert not DocumentURI.called
+    assert not db.add.called
+
+
+@create_or_update_document_uri_fixtures
+def test_create_or_update_document_uri_logs_warning_if_document_ids_differ(
+        log,
+        DocumentURI):
+    """
+    It should log a warning on Document objects mismatch.
+
+    If there's an existing DocumentURI and its .document property is different
+    to the given document it shoulg log a warning.
+
+    """
+    # existing_document_uri.document will not be equal to the given document'
+    existing_document_uri = mock.Mock(document=mock_document())
+    DocumentURI.query.filter.return_value.first.return_value = existing_document_uri
+
+    document.create_or_update_document_uri(
+        db=mock_db(),
+        claimant='http://example.com/example_claimant.html',
+        uri='http://example.com/example_uri.html',
+        type='self-claim',
+        content_type=None,
+        document=mock_document(),
+        created=datetime.datetime.now(),
+        updated=datetime.datetime.now())
+
+    assert log.warn.call_count == 1
+
+
+@create_or_update_document_uri_fixtures
+def test_create_or_update_document_uri_updates_updated_time(DocumentURI):
+    # existing_document_uri has an older .updated time than than the updated
+    # argument we will pass to create_or_update_document_uri().
+    now = datetime.datetime.now()
+    yesterday = now - datetime.timedelta(days=1)
+    existing_document_uri = mock.Mock(updated=yesterday)
+
+    DocumentURI.query.filter.return_value.first.return_value = (
+        existing_document_uri)
+
+    document.create_or_update_document_uri(
+        db=mock_db(),
+        claimant='http://example.com/example_claimant.html',
+        uri='http://example.com/example_uri.html',
+        type='self-claim',
+        content_type=None,
+        document=mock_document(),
+        created=now - datetime.timedelta(days=3),
+        updated=now)
+
+    assert existing_document_uri.updated == now
+
+
+@pytest.mark.usefixtures('DocumentMeta',
+                         'log')
+class TestCreateOrUpdateDocumentMeta(object):
```

Comment:
Is there a reason why the tests for `create_or_update_document_meta` are in a test class, but the tests for `create_or_update_document_uri` are not?
Would be nice if we could keep them consistent (I personally prefer the test class, but I'm fine if you want to stick to the other one).